core/tests/test_apps.ethereum.sign_typed_data.py

```
@@ -19,251 +19,242 @@
+
+            await self.write(msg)
+            del msg
+            return await self.read((expected_type.MESSAGE_WIRE_TYPE,), expected_type)
+
+    # Helper functions from trezorctl to build expected type data structures
+    # TODO: it could be better to group these functions into a class, to visibly differentiate it
+    def get_type_definitions(types: dict) -> dict:
+        result = {}
+        for struct, fields in types.items():
+            members = []
+            for name, type in fields:
```

Ruff (0.15.13)

[error] Variable `type` is shadowing a Python builtin

(A001)